envisage/plugins/ipython_kernel/tests/test_internal_ipkernel.py

```diff
@@ -15,59 +19,106 @@
 
 
 if ipykernel_available:
-    from ipykernel.iostream import IOPubThread
-    from ipykernel.kernelapp import IPKernelApp
+    import ipykernel.iostream
+    import ipykernel.ipkernel
+    import ipykernel.kernelapp
+    import ipykernel.zmqshell
+    import IPython.utils.io
+    import zmq
 
     from envisage.plugins.ipython_kernel.internal_ipkernel import (
         InternalIPKernel)
 
 
+# Machinery for catching logged messages in tests (both for testing
+# the existence and content of those messages, and for hiding the
+# logger output during the test run).
+
+class ListHandler(logging.Handler):
+    """
+    Simple logging handler that stores all records in a list.
+
+    Used by catch_logs below.
+    """
+    def __init__(self):
+        logging.Handler.__init__(self)
+        self.records = []
+
+    def emit(self, record):
+        self.records.append(record)
+
+
 @contextlib.contextmanager
-def temp_filename(filename):
-    tmpdir = tempfile.mkdtemp()
+def catch_logs(logger_name, level=logging.WARNING):
```

Comment:
Unrelated comment - i've seen quite a few `catch_logs` implementations. I see one in `traits` as well. Maybe update the one in `traits` and make it standard across projects?